File: src/codes/exposed-complexity/py/rfc_facade.py
class AuthService:
    def authenticate(self, user_id: str) -> bool:
        print(f"ユーザー認証: {user_id}")
        return True

class UserProfileService:
    def load(self, user_id: str) -> dict:
        print(f"プロフィール取得: {user_id}")
        return {"name": "Hiroshi", "id": user_id}

class AuditService:
    def log(self, action: str):
        print(f"[監査ログ] {action}")

# Facade
class UserProfileFacade:
    def __init__(self):
        self.auth = AuthService()
        self.profile = UserProfileService()
        self.audit = AuditService()

    def show(self, user_id: str):
        if self.auth.authenticate(user_id):
            profile = self.profile.load(user_id)
            print(profile)
            self.audit.log("プロフィール表示")

# 利用例
facade = UserProfileFacade()
facade.show("user-123")

File: src/codes/exposed-complexity/ts/rfc_facade.ts
class AuthService {
  authenticate(userId: string): boolean {
    console.log(`ユーザー認証: ${userId}`);
    return true;
  }
}

class UserProfileService {
  load(userId: string): object {
    console.log(`プロフィール取得: ${userId}`);
    return { name: "Hiroshi", id: userId };
  }
}

class AuditService {
  log(action: string) {
    console.log(`[監査ログ] ${action}`);
  }
}

// Facade
class UserProfileFacade {
  private auth = new AuthService();
  private profile = new UserProfileService();
  private audit = new AuditService();

  show(userId: string) {
    if (this.auth.authenticate(userId)) {
      const profile = this.profile.load(userId);
      console.log(profile);
      this.audit.log("プロフィール表示");
    }
  }
}

// 利用例
const facade = new UserProfileFacade();
facade.show("user-123");
